graph_weather/models/gencast/denoiser.py

```diff
@@ -0,0 +1,209 @@
+"""Denoiser."""
```

Comment:
Could you write a quick comment what the denoiser does?

Reply:
Done!